### ISSUE TITLE
Stopped fetching

### ISSUE THREAD

**Issue by bekoli** [jackjamieson2/yarns-microsub-server#95]
Quick note:Yarns Microsub Server stopped fetching feeds ~3 days ago.

No idea what happened. I enabled WP_DEBUG and checked the log. Got nothing suspicious.

Cleaned my channels and removed some streams that never fetched (beside in preview).

Nothing on the old channels.
A new added feed fetched tho.

Enabled Debug on the Plugin Options page and hit Polling manually. Shortly after I got results in my readers again.

I suspect that some CRON job was removed. How to reinstall this? I'm afraid that it removes all channels when the plugin gets deactivated.

**Comment by jackjamieson2** [comment]
Thanks @bekoli,

The good news: deactivating and/or deleting Yarns will not delete your channels and subscriptions.  It will remove the saved posts, but those will be fetched again once you reinstall.  If it's a cron issue, I think that should resolve it

Before you do that, If you get a chance, would you mind helping with some diagnosis?  
If you install the plugin [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/) you can use it to see what cron jobs are active etc.  

With WP Crontrol installed, go to **Tools**->**Cron events** in the WordPress dashboard.  You should see an event with the hook name `yarns_microsub_server_cron`, an action `Yarns_Microsub_Aggregator::poll()`, and a next run time.

If there is no `yarns_microsub_server_cron` item then it's somehow been removed, otherwise can you let me know the next run time?

**Comment by bekoli** [comment]
Well I can confirm that no further cron job was issued after I manually polled for new feeds.

I use `wp-cli`:

```
+------------------------------------+---------------------+----------------------+------------+
| hook                               | next_run_gmt        | next_run_relative    | recurrence |
+------------------------------------+---------------------+----------------------+------------+
| wp_privacy_delete_old_export_files | 2019-09-30 17:49:48 | 6 minutes 43 seconds | 1 hour     |
| wp_version_check                   | 2019-09-30 23:26:55 | 5 hours 43 minutes   | 12 hours   |
| wp_update_plugins                  | 2019-09-30 23:26:55 | 5 hours 43 minutes   | 12 hours   |
| wp_update_themes                   | 2019-09-30 23:26:55 | 5 hours 43 minutes   | 12 hours   |
| wp_scheduled_delete                | 2019-10-01 10:57:09 | 17 hours 14 minutes  | 1 day      |
| wp_scheduled_auto_draft_delete     | 2019-10-01 14:28:24 | 20 hours 45 minutes  | 1 day      |
| delete_expired_transients          | 2019-10-01 16:11:10 | 22 hours 28 minutes  | 1 day      |
| recovery_mode_clean_expired_keys   | 2019-10-01 17:31:14 | 23 hours 48 minutes  | 1 day      |
+------------------------------------+---------------------+----------------------+------------+
```

Deactivating resulted in a gateway timeout.

> [error] 501#501: *312160 upstream timed out (110: Connection timed out) while reading response header from upstream,

The job `plugin deactivate yarns-microsub-server` from wp-cli never comes back. Machine is in idle. No spike in mem or cpu usage to speak of:

> Debug (bootstrap): Running command: plugin deactivate (0.231s)

The plugin can not be deactivated.

**Comment by jackjamieson2** [comment]
Thanks for these details. I don't know why the cron would have been removed in the first place. Manual polling won't solve that problem because it just runs the poll function directly rather than changing the cron job.

As for the deactivation, I've had an issue in the past where deactivating (from the WordPress dashboard, not cli) displayed a timeout error.  That happened when there were a lot of posts to clean up at once.

However in every case, after returning that error the deactivation still proceeded correctly.  If you're not seeing any increase in memory or cpu usage it might be a different problem, however. 

Overall, this is a tricky one and I'll try to troubleshoot it.

---
Hacky temporary solution:

If you delete the Yarns plugin folder directly, it will be deactivated when you next load your site. 

At that point, reinstalling and activating should create a new cron job.

---

Notes for myself:
1. Why is the cron job being removed?
2. Clicking manual poll could (a) check if the cron job still exists, (b) if so, reset the countdown for this cron, (c) if not, create it again
3. Deactivation should never get stalled. This could be addressed with more efficient processes for removing fetched posts.  Could be implemented by #82

**Comment by bekoli** [comment]
I left it running and it came back after some hours:

```
Debug (bootstrap): Running command: plugin deactivate (0.221s)
Warning: Could not deactivate the 'yarns-microsub-server' plugin.
Success: Deactivated 1 of 1 plugins.
```
Activating it again took just a second.

It's now in cron:

> | yarns_microsub_server_cron         | 2019-09-30 20:02:11 | 14 minutes 43 seconds | 15 minutes |
